plugins/wporg-5ftf/includes/pledge-form.php

```diff
@@ -292,3 +249,101 @@ function parse_contributors( $contributors ) {
 
 	return $sanitized_contributors;
 }
+
+/**
+ * Check the submission for valid data.
+ *
+ * @return false|WP_Error Return any errors in the submission, or false if no errors.
+ */
+function check_invalid_submission( $submission ) {
+	$has_required = PledgeMeta\has_required_pledge_meta( $submission );
+	if ( is_wp_error( $has_required ) ) {
+		return $has_required;
+	}
+
+	$email = sanitize_meta(
+		PledgeMeta\META_PREFIX . 'org-pledge-email',
+		$submission['org-pledge-email'],
+		'post',
+		Pledge\CPT_ID
+	);
+
+	if ( has_existing_pledge( $email, 'email' ) ) {
+		return new WP_Error(
+			'existing_pledge_email',
+			__( 'This email address is already connected to an existing pledge.', 'wporg' )
+		);
+	}
+
+	$domain = PledgeMeta\get_normalized_domain_from_url( $submission['org-url'] );
+
+	if ( has_existing_pledge( $domain, 'domain' ) ) {
+		return new WP_Error(
+			'existing_pledge_domain',
+			__( 'A pledge already exists for this domain.', 'wporg' )
+		);
+	}
+
+	return false;
+}
+
+/**
+ * Upload the logo image into the media library.
+ *
+ * @param array $logo $_FILES array for the uploaded logo.
+ * @return int|WP_Error Upload attachment ID, or WP_Error if there was an error.
+ */
+function upload_image( $logo ) {
+	if ( ! $logo ) {
+		return false;
+	}
+
+	// Process image.
+	if ( ! function_exists('media_handle_upload') ) {
+		require_once ABSPATH . 'wp-admin/includes/image.php';
+		require_once ABSPATH . 'wp-admin/includes/file.php';
+		require_once ABSPATH . 'wp-admin/includes/media.php';
+	}
+
+	if ( ! function_exists('check_upload_size') ) {
+		require_once ABSPATH . 'wp-includes/ms-functions.php';
+		require_once ABSPATH . 'wp-admin/includes/ms.php';
+	}
+
+	add_filter( 'upload_mimes', __NAMESPACE__ . '\safelist_image_mimes' );
+	add_filter( 'pre_site_option_fileupload_maxk', __NAMESPACE__ . '\restrict_file_size' );
+	add_filter( 'wp_handle_sideload_prefilter', 'check_upload_size' );
+
+	$logo_id = \media_handle_sideload( $logo, 0 );
+
+	remove_filter( 'upload_mimes', __NAMESPACE__ . '\safelist_image_mimes' );
+	remove_filter( 'pre_site_option_fileupload_maxk', __NAMESPACE__ . '\restrict_file_size' );
+	remove_filter( 'wp_handle_sideload_prefilter', 'check_upload_size' );
+
+	return $logo_id;
+}
+
+/**
+ * Only allow image mime types.
+ *
+ * @param array $mimes Mime types keyed by the file extension regex corresponding to those types.
+ */
+function safelist_image_mimes( $mimes ) {
+	return array(
+		'jpg|jpeg|jpe' => 'image/jpeg',
+		'gif'          => 'image/gif',
+		'png'          => 'image/png',
+		'bmp'          => 'image/bmp',
+		'tiff|tif'     => 'image/tiff',
+		'ico'          => 'image/x-icon',
```

Comment:
`MAYBE` - Very minor, but do we want to allow `bmp`/`tiff`/`ico`? It doesn't seem like those would be good formats for this context.